### PR TITLE
Add GitHub Actions script for releases on 3.0

### PR DESCRIPTION
**Pull request by krashish8**
Changes proposed in this pull request:
- Ports #1855 to 3.0

@pgRouting/admins

### CODE REVIEW

**Comment by cvvergara** [comment]
@krashish8 
This comment aaplies to all 3 PR
A requirement was changed,
Than needs to be informed on:
README.md
and it would also affect CMakeLists, but you can write on the README.md that the min sphinx version change will be enforced on 3.2.0

Did you test it in your fork?, did you create a new tag in your fork?

**Comment by krashish8** [comment]
@cvvergara 
Yes, I had tested this on my fork but I later deleted the tags, releases and commits on gh-pages.
I again created the tags on my fork for v3.0.5 and v3.1.3: https://github.com/krashish8/pgrouting/releases
The `gh-pages` branch is also updated on my fork by GitHub Actions: https://github.com/krashish8/pgrouting/commits/gh-pages

**v3.0.5**
GitHub Action: https://github.com/krashish8/pgrouting/runs/1752864273?check_suite_focus=true

The actions create a "draft" release, so that we can have a confirmation from our side, before actually publishing the release.
I published only the v3.1.3 release, the v3.0.5 is still a draft (Draft release is visible to only collaborators, so it might not be visible to you). The draft release looks like this:
image


**v3.1.3** (created from the master branch on my fork, so that developers documentation is also updated on `gh-pages`).
GitHub Action: https://github.com/krashish8/pgrouting/runs/1752865289?check_suite_focus=true

**Comment by krashish8** [comment]
Regarding the pdf, I was getting some latex errors/warnings, so I haven't updated that.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+          sudo apt-get install -y \
+            libboost-graph-dev \
+            postgresql-${PGVER} \
+            postgresql-${PGVER}-postgis-${PGIS} \
+            postgresql-${PGVER}-postgis-${PGIS}-scripts \
+            postgresql-server-dev-${PGVER} \
+            graphviz \
+            doxygen
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
```
Comment (cvvergara) [inline .github/workflows/release.yml:74]: on 3.0 & 3.1
Changing
```
pip install -r requirements.txt
```
to
```
pip install Sphinx
pip install sphinx-bootstrap-theme
```
& 
Roll back the changes on README & installation

**Review by cvvergara — CHANGES_REQUESTED** [review]

**Review by cvvergara — APPROVED** [review]

**Pull request merged**